crates/sui-replay/src/replay.rs

```diff
@@ -801,64 +792,58 @@ impl LocalExec {
             pre_run_sandbox.transaction_info.chain,
         );
         let required_objects = pre_run_sandbox.required_objects.clone();
+        let store = InMemoryStorage::new(required_objects.clone());
         let shared_object_refs = pre_run_sandbox.transaction_info.shared_object_refs.clone();
 
-        assert_eq!(
-            pre_run_sandbox
-                .transaction_info
-                .sender_signed_data
-                .intent_message()
-                .intent,
-            Intent::sui_transaction()
+        let transaction =
+            Transaction::new(pre_run_sandbox.transaction_info.sender_signed_data.clone());
+
+        let new_tx_digest = transaction.digest();
+        let mut shared_locks = InMemorySharedLocks::new();
+        shared_locks.add_shared_locks(
+            TransactionKey::Digest(*new_tx_digest),
+            shared_object_refs
+                .iter()
+                .map(|(id, version, _)| (*id, *version))
+                .collect(),
         );
```

Comment:
The rest of this PR makes sense to me, but I don't really understand what this section is doing -- so that gives me pause to stamp. It seems like we create this data structure, add some stuff to it, and then don't do anything with that.

Reply:
Good catch! That was a failed attempt to share code. Let me remove them